src/index.ts

```diff
@@ -61,16 +60,19 @@ export const useWallet: UseWallet = () => {
     web3ModalProvider.on('disconnect', () => {
       web3Modal.clearCachedProvider();
```

Comment:
Should this also unset state?

Reply:
I don't think it's caused any issue yet but don't think it'd hurt either. Maybe we just call the disconnect() function directly there?

Reply:
That's what I'm doing now locally. I think I encountered an issue when jumping around my wallet and trying different things where the wallet was definitely disconnected but some of the state values here still seemed cached.